src/utils/openmc/tallies.py

```diff
@@ -0,0 +1,1239 @@
+
+        if not nuclide in self._nuclides:
+            msg = 'Unable to remove nuclide {0} from Tally ID={1} since the ' \
+                  'Tally does not contain this nuclide'.format(nuclide, self._id)
+            ValueError(msg)
+
+        self._nuclides.remove(nuclide)
+
+
+    def compute_std_dev(self, t_value=1.0):
+
+        # Calculate sample mean and standard deviation
+        self._mean = self._sum / self._num_realizations
+        self._std_dev = np.sqrt((self._sum_sq / self._num_realizations - \
+                                 self._mean**2) / (self._num_realizations - 1))
+        self._std_dev *= t_value
+
+
+    def __repr__(self):
+
+        string = 'Tally\n'
+        string += '{0: <16}{1}{2}\n'.format('\tID', '=\t', self._id)
+        string += '{0: <16}{1}{2}\n'.format('\tName', '=\t', self._label)
+
+        string += '{0: <16}\n'.format('\tFilters')
+
+        for filter in self._filters:
+            string += '{0: <16}\t\t{1}\t{2}\n'.format('', filter._type,
+                                                          filter._bins)
+
+        string += '{0: <16}{1}'.format('\tNuclides', '=\t')
+
+        for nuclide in self._nuclides:
+            string += '{0} '.format(nuclide._name)
+
+        string += '\n'
+
+        string += '{0: <16}{1}{2}\n'.format('\tScores', '=\t', self._scores)
+        string += '{0: <16}{1}{2}\n'.format('\tEstimator', '=\t', self._estimator)
+
+        return string
+
+
+    def get_num_filters(self):
+        return len(self._filters)
+
+
+    def get_num_filter_bins(self):
+
+        num_bins = 1
+
+        for filter in self._filters:
+            num_bins *= filter.get_num_bins()
+
+        return num_bins
+
+
+    def get_num_nuclides(self):
+        return len(self._nuclides)
+
+
+    def get_num_scores(self):
+        return len(self._scores)
+
+
+    def get_num_score_bins(self):
+        return self._num_score_bins
+
+
+    def get_num_bins(self):
+        num_bins = self.get_num_filter_bins()
+        num_bins *= self.get_num_nuclides()
+        num_bins *= self.get_num_score_bins()
+        return num_bins
+
+
+    def get_tally_xml(self):
+
+        element = ET.Element("tally")
+
+        # Tally ID
+        element.set("id", str(self._id))
+
+        # Optional Tally label
+        if self._label != '':
+            element.set("label", self._label)
+
+        # Optional Tally filters
+        for filter in self._filters:
+
+            subelement = ET.SubElement(element, "filter")
+            subelement.set("type", str(filter._type))
+
+            if not filter._bins is None:
+
+                bins = ''
+                for bin in filter._bins:
+                    bins += '{0} '.format(bin)
+
+                subelement.set("bins", bins.rstrip(' '))
+
+        # Optional Nuclides
+        if len(self._nuclides) > 0:
+
+            nuclides = ''
+            for nuclide in self._nuclides:
+                nuclides += '{0} '.format(nuclide._name)
```

Comment:
This error was related to the one above. I just pushed a conditional check for a ``openmc.Nuclide`` object vs a "total" string (or any string, such that the user may now directly pass in a nuclide string name "U-235" via the ``tally.add_nuclide(...)`` routine rather than a ``openmc.Nuclide`` object if they wish).